### PR TITLE
Removing dependency on onos-config

### PR DESCRIPTION
**Pull request by SeanCondon**
I'm taking away the legacy structs and walking the YANG schema to directly define the proto based `[]*admin.ReadOnlyPath` and `[]*admin.ReadWritePath`

I've split the previous utils/path.go in to `path/values.go` and `path/extract.go`

Also adding some unit tests and resurrecting some ones from older tags

### CODE REVIEW

File: pkg/path/extract.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+			name += fmt.Sprintf("[%s=*]", k)
+		}
+	} else {
+		name = fmt.Sprintf("%s/%s", parentAndSubPath, dirEntry.Name)
+	}
+
+	return name
+}
+
+func toValueType(entry *yang.YangType, isLeafList bool) (configapi.ValueType, []uint64, error) {
```
Comment (teone) [inline pkg/path/extract.go:240]: I do have pretty much the same method here: https://github.com/onosproject/config-models/blob/master/pkg/gnmi-client-gen/utils.go#L90

We should try to consolidate at a certain point

**Review by teone — APPROVED** [review]
Looks good some minor comments.

While looking I added some unit tests to understand the `formatName` function, they are attached in case you want to add them:
[formatName.diff.txt](https://github.com/onosproject/config-models/files/8834067/formatName.diff.txt)

File: pkg/path/extract.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+	var name string
+	if isList {
+		//have to ensure index order is consistent where there's more than one
+		keyParts := strings.Split(dirEntry.Key, " ")
+		sort.Slice(keyParts, func(i, j int) bool {
+			return keyParts[i] < keyParts[j]
+		})
+		name = fmt.Sprintf("%s/%s", parentAndSubPath, dirEntry.Name)
+		for _, k := range keyParts {
+			name += fmt.Sprintf("[%s=*]", k)
```
[the remark this answers is not in the document]
Comment (SeanCondon) [reply pkg/path/extract.go:231]: no it just represents the name of the key

File: pkg/path/extract.go (hunk 1 above)
Comment (SeanCondon) [reply pkg/path/extract.go:240]: This was written a long time ago here https://github.com/onosproject/onos-config/blob/be9fe145adaece61bb521b4d3817b1108214b471/pkg/utils/path/path.go#L370

**Pull request merged**